packages/pluggableWidgets/line-chart-native/src/utils/SeriesDataLoadingFunction.ts

```diff
@@ -0,0 +1,203 @@
+import { ensure } from "@mendix/pluggable-widgets-tools";
+import { Big } from "big.js";
+import { ObjectItem } from "mendix";
+
+import { SeriesType } from "../../typings/LineChartProps";
+import { LineChartDataPoints, LineChartSeries } from "../components/LineChart";
+
+interface DataSourceItemGroup {
+    groupByAttributeValue: string | boolean | Date | Big;
+    dynamicSeriesNameValue?: string;
+    dynamicStylePropertyNameValue?: string;
+    items: ObjectItem[];
+}
+
+export function loadSeries(series: SeriesType[]): LineChartSeries[] | null {
```

Comment:
Cant we create a hook instead? `useSeries` then in your main class you dont need to use useEffect

Reply:
Created a hook. Awesome to do this for the first time! :)